src/plugins/undoRedo/undoRedo.js

```diff
@@ -428,39 +427,48 @@ UndoRedo.RemoveColumnAction.prototype.undo = function(instance, undoneCallback)
   const ascendingIndexes = this.indexes.slice(0).sort();
   const sortByIndexes = (elem, j, arr) => arr[this.indexes.indexOf(ascendingIndexes[j])];
 
+  const removedDataLength = this.data.length;
   const sortedData = [];
-  rangeEach(this.data.length - 1, (i) => {
-    sortedData[i] = arrayMap(this.data[i], sortByIndexes);
-  });
 
-  let sortedHeaders = [];
-  sortedHeaders = arrayMap(this.headers, sortByIndexes);
+  for (let rowIndex = 0; rowIndex < removedDataLength; rowIndex++) {
+    sortedData.push(arrayMap(this.data[rowIndex], sortByIndexes));
+  }
 
+  const sortedHeaders = arrayMap(this.headers, sortByIndexes);
+  const isFormulaPluginEnabled = instance.getPlugin('formulas') ? instance.getPlugin('formulas').enabled : false;
```

Comment:
I don't know whether it is cleaner 🤔.

```suggestion
  const isFormulaPluginEnabled = instance.getPlugin('formulas')?.enabled || false;
```

Reply:
Fixed within https://github.com/handsontable/handsontable/pull/6930/commits/fea3b26e978e4beeb77ca87d411f21d3e5e3edf9 👍